Code/src/main/java/nl/utwente/group10/haskell/type/VarT.java

```diff
@@ -132,12 +148,12 @@ public VarT getFresh() {
             instance = this.instance.get().getFresh();
         }
 
-        return new VarT(this.name, this.constraints, instance);
+        return new VarT(this.prefix, this.uid, this.constraints, instance);
```

Comment:
If you return a new VarT with the same uid, is that uid still a unique identifier?